tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -4075,7 +4065,26 @@ void VerifyProgOffsetResults(unsigned *pPixels, bool bCheckDeriv) {
   // Check that each element matches the expected value given the offset
   unsigned ix = 0;
   int coords[18] = {100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700, 750, 800, 850, 900, 950};
-  int offsets[18] = {CLAMPOFFSET(-9), -8, -7, -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, CLAMPOFFSET(8)};
+
+  int offsets[18] = {CLAMPOFFSET((unsigned)-9),
+                     -8,
+                     -7,
+                     -6,
+                     -5,
+                     -4,
+                     -3,
+                     -2,
+                     -1,
+                     0,
+                     1,
+                     2,
+                     3,
+                     4,
+                     5,
+                     6,
+                     7,
+                     CLAMPOFFSET(8)};
```

Comment:
My mistake. With the small change to the first element, my clang-format does the same thing. I really dislike this, but at least it's accurate.

Reply:
Well, if you agree that this is ugly and don't mind changing the code to satisfy the tool, you could move the cast into the macro since it's only used here: 
``` c++
#define CLAMPOFFSET(offset) (((unsigned)(offset)<<28)>>28)
```